components/script/dom/htmlelement.rs

```diff
@@ -272,9 +272,13 @@ impl HTMLElementMethods for HTMLElement {
 
     // https://drafts.csswg.org/cssom-view/#dom-htmlelement-offsetparent
     fn GetOffsetParent(&self) -> Option<Root<Element>> {
-        if self.is::<HTMLBodyElement>() || self.is::<HTMLHtmlElement>() {
+        if self.is::<HTMLHtmlElement>() {
```

Comment:
Should be better to compare self to its document's root element, rather than go up the whole hierarchy.